### PR TITLE
feat: Use LazyDocs to generate markdown files, based on docstrings

### PR DESCRIPTION
**Pull request by SamVanhoutte**
Intention is to automatically generate .md files during the CI build and inject those files in the documentation website.  
Will verify with @Mcalewaert on the actual process

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,14 +5,56 @@ Python Machine Learning development in a breeze.
...
+
+# Local development
+    
+It can be quite common that you are using the arcus-ml or arcus-azureml packages on other projects and you need some changes or additional functionality being added to the package.  Obviously, it's possible to follow the entire release pipeline (make a PR, get it approved and merged, wait for package to appear on the PyPi feed and upgrade the package).  This workflow is too tedious and will not increase your productivity.
+
+The approach to go, is to leverage the following command, which will add a symbolic link to your development directory from the python packages.  That way, you always refer to the latest code that is on your development environment.  It is advised, though not required to leverage virtual environment (like with conda) for this.
+
+```shell
+pip install -e /path-to-arcus
+```
```
Comment (stijnmoreels) [inline README.md:45]: Already a good first step to having some docs on local development. I would love to see some more practical information.
* Link to PyPi feed
* Link to how to set such a virtual environment

File: README.md [hunk 2]
```diff
@@ -5,14 +5,56 @@ Python Machine Learning development in a breeze.
...
+
+The actual build process executes the following steps:
+
+1. Update package version, based on build
+2. Install following python packages: `pip`, `setuptools`, `wheel`, `pdoc3` and `lazydocs`
+3. Install the prerequisites from the `requirements.txt` file
+4. Run the unit tests, found in the solution
+5. Publish the test & code coverage results
+6. Build & push the packages (wheels)
+7. Generate the code documentation, using lazydocs
```
Comment (stijnmoreels) [inline README.md:57]: This is a good first draft of the build process, but I think we need some work here. The steps describe an abstracted task, but are missing maybe the actual commends/low-level steps one has to take. Keep in mind that we should be accessible for both beginner/advanced developers.

**Review by stijnmoreels — CHANGES_REQUESTED** [review]
👍 Good first step, let's see how we can improve.

File: README.md (hunk 1 above)
Comment (SamVanhoutte) [reply README.md:45]: Created #98 for this.  Focus of this PR should be on the actual build + doc generation.

File: README.md (hunk 2 above)
Comment (SamVanhoutte) [reply README.md:57]: thanks.
Created #98 for this.  Focus of this PR should be on the actual build + doc generation.

**Comment by SamVanhoutte** [comment]
@stijnmoreels - could you shed your light on the following issue I have? 
so, in this PR I have changed the CI build to generate markdown files and git commit them to the branch that is being merged with the PR... 
after lot's of tries, this works.
but I want to avoid that making that commit, triggers a new CI build, that triggers a new CI build, etc, etc (endless builds loop)

I thought just by adding the following yaml lines in the build file, would avoid triggering a new build, but apparently it doesn't.  would you have an idea? 

```yaml
trigger:
  branches:
    include:
    - master
  paths:
    include:
    - arcus/*
    - tests/*
    - requirements.txt
    exclude:
    - docs/preview/*

pr:
  paths:
    include:
    - arcus/*
    - tests/*
    - build/ci-build.yml
    - requirements.txt
    exclude:
    - docs/preview/*
```

**Comment by SamVanhoutte** [comment]
Will close PR (too many commits, due to endless loop of CI-builds)
Current solution is to locally run the unit tests, which generate the documentation and commit them with the changes

**Review by ALaks96 — COMMENTED** [review]
Great stuff! thanks sam

**Pull request closed without merging**